### PR TITLE
tests: unit: maintainers_test: Add failing tests

### PR DESCRIPTION
**Pull request by Lincoln-Yuji**
The kw 'maintainers' feature lacks tests for its multiple corner cases where it should fail. Adding such tests ensures a larger test coverage for expected behaviors of this feature, possibly preventing future accidental undesirable changes.

### CODE REVIEW

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi @Lincoln-Yuji ,
Thanks for this PR, I added some minor comments. Btw, this test is very old and probably requires some code refactor (e.g., split tests in small functions) and code style adjustment. Additionally, you can check the coverage report to see which part of the code has not been tested yet:

https://app.codecov.io/gh/kworkflow/kworkflow/blob/unstable/src%2Fmaintainers.sh

File: tests/unit/maintainers_test.sh [hunk 1]
```diff
@@ -76,6 +77,8 @@ function oneTimeSetUp()
 function oneTimeTearDown()
 {
   rm -rf "$FAKE_KERNEL"
+  rm -rf "$SHUNIT_TMPDIR"
+  mkdir -p "$SHUNIT_TMPDIR"
```
Comment (rodrigosiqueira) [inline tests/unit/maintainers_test.sh:81]: https://kworkflow.org/content/codingstyle.html#avoid-short-options

Also, before every rm command, we must check if the variable is not empty or pointing to a weird path. This is very important to keep the code safe and avoid catastrophic removals. Check the function is_safe_path_to_remove.

Also, I know the old code just uses rm, but that is legacy code with some bad practices, and it must be updated.

File: tests/unit/maintainers_test.sh [hunk 2]
```diff
@@ -109,16 +112,157 @@ function test_maintainers_main()
...
   ret="$(maintainers_main .)"
-  multilineAssertEquals "$CORRECT_TMP_FS_MSG" "$ret"
+  multilineAssertEquals "$ret" "$CORRECT_TMP_FS_MSG"
+
+  cd "$original_dir" || {
+    fail "(${LINENO}) It was not possible to move back from temp directory"
+    return
+  }
+  rm -rf "$SHUNIT_TMPDIR"
+  mkdir -p "$SHUNIT_TMPDIR"
```
Comment (rodrigosiqueira) [inline tests/unit/maintainers_test.sh:129]: Maybe this should be in the tearDown function, and we cleanup everything per test.

File: tests/unit/maintainers_test.sh [hunk 3]
```diff
@@ -109,16 +112,157 @@ function test_maintainers_main()
...
   multilineAssertEquals "$CORRECT_TMP_MSG" "$ret"
 
   ret="$(maintainers_main .)"
-  multilineAssertEquals "$CORRECT_TMP_FS_MSG" "$ret"
+  multilineAssertEquals "$ret" "$CORRECT_TMP_FS_MSG"
+
+  cd "$original_dir" || {
+    fail "(${LINENO}) It was not possible to move back from temp directory"
+    return
+  }
```
Comment (rodrigosiqueira) [inline tests/unit/maintainers_test.sh:127]: Maybe you can move this `cd` to a specific folder to the `setUp` function and go back to the `tearDown` function; that way, each test can be simplified.

File: tests/unit/maintainers_test.sh [hunk 4]
```diff
@@ -109,16 +112,157 @@ function test_maintainers_main()
...
+{
+  local ret
+  local -r original_dir="$PWD"
+
+  cd "$FAKE_KERNEL" || {
+    fail "(${LINENO}) It was not possible to move to temporary directory"
+    return
+  }
+
+  msg="$(maintainers_main --update-patch 'fs/some_file')"
```
Comment (rodrigosiqueira) [inline tests/unit/maintainers_test.sh:179]: Declare this variable as local.
Also, use a more intuitive name, something like output_from_function_xpto...

P.S.: Please review this entire commit with this comment in mind.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

**Comment by rodrigosiqueira** [comment]
> Hello, @rodrigosiqueira we have finished applying your requested changes and we are now waiting for new feedbacks. Thank you very much for reviewing our code and suggesting some changes.
> 
> Regarding the string concatenation style for `LINENO`, we ended up applying it for all occurences of this pattern in this file, not only the new lines we made. We did this to keep it consistent accross the entire file.
> 
> However, we had some issues adding quotes to this line:
> 
> ```shell
> cp -f tests/unit/samples/update_patch_test{_model,}{,2}.patch "$FAKE_KERNEL"/
> ```
> 
> It seems that adding either single or double quotes makes this command not copy the patch files and we are not sure if this expression is expanded by the shell itself or if the `cp` command handles this format.
> 
> Because of that we decided to only add the quotes to the new line we made and leave the others untouched.

Hi @Lincoln-Yuji , ignore my comment about cp.

**Comment by Lincoln-Yuji** [comment]
Hi @rodrigosiqueira, we've finished applying your new suggestions and we hope everything is fine now. Thank you for the feedback

**Comment by rodrigosiqueira** [comment]
Thanks! Change merged into the unstable branch.
